File: project1_prob4/src/com/company/Animal.java
package com.company;

public class Animal {
    private String name;
    private String type;
    public Animal(String animalName){
        name = animalName;
    }
    public String wakeup(){
        return "Animal: wake up";
    }

    public String makeNoise(){
        return "Animal: make noise. ";
    }

    public String eat(){
        return "Animal: eat. ";
    }

    public String roam(){
        return "Animal: roam. ";
    }

    public String sleep(){
        return "Animal: sleep.";
    }

    public String getName() {
        return name;
    }

    public String getType(){
        return type;
    }


}

File: project2_Q1/animal.py
import abc

class Animal(object):
    
    _make_noise_strategy = None 
    animal_type = None
    
    def __init__(self, name):
        self.name = name
    
    def makeNoise(self):
        self._make_noise_strategy.makeNoise()        

    def wakeup(self):
        return 'Animal: wake up.'
    
    def eat(self):
        return 'Animal: eat.'
    
    def roam(self):
        return 'Animal: roam.'
    
    def sleep(self):
        return 'Animal: sleep.'
        
    def getName(self):
        return str(self.name)
    
    def getType(self, animalType):
        return str(self.animalType)
    

